packages/tracing/src/transaction.ts

```diff
@@ -7,6 +7,10 @@ import { Span as SpanClass, SpanRecorder } from './span';
 /** JSDoc */
 export class Transaction extends SpanClass implements TransactionInterface {
   public name: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _metadata: { [key: string]: any } = {};
```

Comment:
We dont use anything other than `string | number`, so we should be able to use it instead of `any` for now.

Reply:
Well... It'd need to be `{ [key: string]: { [key: string]: string | number } }`. At that point I feel like it should just be an interface with named properties.

```
interface TransactionMetadata {
  transactionSampling?: { [key: string]: string | number };
}
```

What do you think?

Reply:
Sounds good to me. Definitely more descriptive.